streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -61,117 +59,48 @@
 
     private final Logger log;
     private final TaskId taskId;
-    private final boolean eosEnabled;
-    private final String applicationId;
+    private final Consumer<byte[], byte[]> mainConsumer;
+    private final StreamsProducer streamProducer;
```

Comment:
We use `StreamsProducer` now that encapsulates the actually producer, does transaction state handling, and error handling. `RecordCollectorImpl` is now agnostic to transactions as desired.